book/src/user_manual/join_from_welcome.md

```diff
@@ -17,3 +17,41 @@ to inspect the message, e.g. to determine the identity of the sender.
 Pay attention not to forward a Welcome message to a client before its associated
 commit has been accepted by the Delivery Service.  Otherwise, you would end up
 with an invalid MLS group instance.
+
+# Examining a welcome message
+
+When a client is invited to join a group, the application can allow the client to decide whether or not to join the group. In order to determine whether to join the group, the application can inspect information provided in the welcome message, such as who invited them, who else is in the group, what extensions are available, and more. If the application decides not to join the group, the welcome must be discarded to ensure that the local state is cleaned up.
+
+After receiving a `MlsMessageIn` from the delivery service, the first step is to extract the `MlsMessageBodyIn`, and determine whether it is a welcome message.
+```rust,no_run,noplayground
+{{#include ../../../openmls/tests/book_code_discard_welcome.rs:not_join_group_welcome}}
+```
+
+The next step is to process the `Welcome`.
+```rust,no_run,noplayground
+{{#include ../../../openmls/tests/book_code_discard_welcome.rs:not_join_group_processed_welcome}}
+```
+
+At this stage, there are some more pieces of information in the `ProcessedWelcome` that could be useful to the application. For example, it can be useful to check which extensions are available. Much of this information can also be checked using the `StagedWelcome` that is produced in the next step.
```

Comment:
I think this maybe should mention that a lot of the values here are not validated yet, and are only validated when turning it into a StagedWelcome.